lib/rules/no-wildcard-postmessage.js

```diff
@@ -9,32 +10,56 @@
 // Rule Definition
 //------------------------------------------------------------------------------
 
-module.exports = function (context) {
-  "use strict";
-  return {
-    CallExpression: function (node) {
-      // postMessage and somewindow.postMessage
-      var funcName;
-      if (node.callee.name) {
-        funcName = node.callee.name;
-      } else if (node.callee.property && node.callee.property.name) {
-        funcName = node.callee.property.name;
-      } else {
-        // anonymous function
-        return;
-      }
-      if (funcName === "postMessage") {
-        if (node.arguments.length > 1) {
-          if ((node.arguments[1].type === "Literal") &&
-            (node.arguments[1].value === "*")) {
-            context.report(node, "Using postMessage() with" +
-              " wildcard targets is not allowed.");
-          }
+module.exports = {
+  meta: {
+    docs: {
+      description: "Flag Wildcard Targets in `postmessage`",
+      recommended: true
+    },
+    fixable: null,
+    schema: [
+      // fill in your rschema
+    ]
+  },
+
+  create(context) {
+    //----------------------------------------------------------------------
+    // Helpers
+    //----------------------------------------------------------------------
+
+    function isLiteral(node) {
+      return node && node.type === "Literal";
+    }
+
+    function isWildcard(node) {
+      return isLiteral(node) && node.value === "*";
+    }
+
+    //----------------------------------------------------------------------
+    // Public
+    //----------------------------------------------------------------------
+
+    return {
+      CallExpression(node) {
+        // postMessage and somewindow.postMessage
+        let funcName;
+        if (node.callee.name) {
+          funcName = node.callee.name;
+        } else if (node.callee.property && node.callee.property.name) {
+          funcName = node.callee.property.name;
         } else {
+          // anonymous function
           return;
         }
 
+        if (funcName === "postMessage") {
+          const [, originNode] = node.arguments;
+
+          if (isWildcard(originNode)) {
```

Comment:
Not sure this is worth the `isWildcard` and `isLiteral` indirection, when you're using both functions just once